### PR TITLE
[INLONG-8823][Manager] Supporting data flow to Pulsar

### LINKED ISSUE #8823: [Feature][Manager] Supporting data flow to Pulsar
### Description

support pulsar dataflow
support pulsar datanode 

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by castorqin**
### Prepare a Pull Request
*(Change the title refer to the following example)*

- Title Example: [INLONG-XYZ][Component] Title of the pull request

*(The following *XYZ* should be replaced by the actual [GitHub Issue](https://github.com/apache/inlong/issues) number)*

- Fixes #8823 

### Motivation

support pulsar dataflow

### Modifications
The details can be found at [#8823 ](https://github.com/apache/inlong/issues/8823).

### CODE REVIEW

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sink/pulsar/PulsarSinkOperator.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    private static final Logger LOGGER = LoggerFactory.getLogger(PulsarSinkOperator.class);
+
+    @Autowired
+    private ObjectMapper objectMapper;
+    @Autowired
+    InlongStreamEntityMapper inlongStreamEntityMapper;
+
+    @Override
+    public Boolean accept(String sinkType) {
+        return SinkType.KAFKA.equals(sinkType);
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sink/pulsar/PulsarSinkOperator.java:61]: ` SinkType.PULSAR.equals(sinkType);`

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sink/pulsar/PulsarSinkOperator.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    }
+
+    @Override
+    protected String getSinkType() {
+        return SinkType.KAFKA;
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sink/pulsar/PulsarSinkOperator.java:66]: `SinkType.PULSAR;`

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/consts/SortStandAloneConfig.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ */
+
+package org.apache.inlong.manager.common.consts;
+
+/**
+ * Constants of sort standalone config.
+ */
+public interface SortStandAloneConfig {
+
+    interface PulsarParamsConfig {
```
Comment (vernedeng) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/consts/SortStandAloneConfig.java:25]: There are two types of params, id params and sink params. Please specify which params of this config is.

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sink/pulsar/PulsarSink.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+@SuperBuilder
+@AllArgsConstructor
+@ToString(callSuper = true)
+@EqualsAndHashCode(callSuper = true)
+@ApiModel(value = "pulsar sink info")
+@JsonTypeDefine(value = SinkType.PULSAR)
+public class PulsarSink extends StreamSink {
+
+    @ApiModelProperty("pulsar service url")
+    private String serviceUrl;
```
Comment (vernedeng) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sink/pulsar/PulsarSink.java:47]: ServiceUrl is the params of Pulsar data_node

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sink/pulsar/PulsarSinkDTO.java [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+ * Pulsar sink info
+ */
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class PulsarSinkDTO {
+
+    @ApiModelProperty("pulsar service url")
+    private String serviceUrl;
```
Comment (vernedeng) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sink/pulsar/PulsarSinkDTO.java:44]: ditto

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sink/pulsar/PulsarSinkOperator.java [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+        Map<String, String> params = super.parse2IdParams(streamSink, fields);
+        PulsarSinkDTO pulsarSinkDTO;
+        try {
+            pulsarSinkDTO = objectMapper.readValue(streamSink.getExtParams(), PulsarSinkDTO.class);
+        } catch (JsonProcessingException e) {
+            LOGGER.error("parse id error", e);
+            return params;
+        }
+        params.put(PulsarParamsConfig.TOPIC, pulsarSinkDTO.getTopic());
+        params.put(PulsarParamsConfig.KEY_SEPARATOR, inlongStreamEntity.getDataSeparator());
```
Comment (vernedeng) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sink/pulsar/PulsarSinkOperator.java:114]: Separator and datatype are the common properties of all sinks, not only Pulsar sink's

**Review by vernedeng — CHANGES_REQUESTED** [review]

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sink/pulsar/PulsarSinkOperator.java (hunk 1 above)
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sink/pulsar/PulsarSinkOperator.java]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sink/pulsar/PulsarSinkOperator.java (hunk 2 above)
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sink/pulsar/PulsarSinkOperator.java]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sink/pulsar/PulsarSinkOperator.java (hunk 6 above)
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sink/pulsar/PulsarSinkOperator.java]: done

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/consts/SortStandAloneConfig.java (hunk 3 above)
Comment (castorqin) [reply inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/consts/SortStandAloneConfig.java]: done

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sink/pulsar/PulsarSink.java (hunk 4 above)
Comment (castorqin) [reply inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sink/pulsar/PulsarSink.java:47]: Creating a Pulsar sink requires support for configuring serviceUrl

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sink/pulsar/PulsarSinkDTO.java (hunk 5 above)
Comment (castorqin) [reply inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sink/pulsar/PulsarSinkDTO.java:44]: Creating a Pulsar sink requires support for configuring serviceUrl

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/consts/SinkType.java [hunk 7]
```diff
@@ -35,4 +35,5 @@ public class SinkType extends StreamType {
     public static final String STARROCKS = "STARROCKS";
     public static final String KUDU = "KUDU";
     public static final String REDIS = "REDIS";
+    public static final String PULSAR = "PULSAR";
```
Comment (fuweng11) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/consts/SinkType.java:38]: StreamType.PULSAR

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/pulsar/PulsarResourceOperator.java [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+            Tenants tenants = pulsarAdmin.tenants();
+            Clusters clusters = pulsarAdmin.clusters();
+            Set<String> allowClusters = new HashSet<>(clusters.getClusters());
+            if (!tenants.getTenants().contains(pulsarSinkDTO.getTenant())) {
+                TenantInfo tenantInfo = TenantInfo.builder().allowedClusters(allowClusters)
+                        .adminRoles(Collections.emptySet()).build();
+                tenants.createTenant(pulsarSinkDTO.getTenant(), tenantInfo);
+            }
+            List<String> namespaces = pulsarAdmin.namespaces().getNamespaces(pulsarSinkDTO.getTenant());
+            if (namespaces == null || namespaces.isEmpty() || !namespaces.contains(pulsarSinkDTO.getNamespace())) {
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/pulsar/PulsarResourceOperator.java:97]: CollectionUtils.isNotEmpty()

**Review by vernedeng — APPROVED** [review]
LGTM

**Review by luchunliang — APPROVED** [review]

**Review by fuweng11 — APPROVED** [review]

**Pull request merged**